### PR TITLE
feat(api): remove layout.json; fix gradle error

### PR DESCRIPTION
**Pull request by cbeach47**
History in #40 

Currently, `widgetType` is required.  `generic` and `portlet` works for values, though I haven't teased out what the possible values are.

`widgetUrl` can be passed in, but is not immediately clear the use case, so it's not be tested.

For general use, include parameters in your portlet definition of `widgetTemplate` and `widgetType`.

Confirmed the personalization efforts from uPortal now come through in the `context-nav-menu`.

I still need to do some cross browser testing inside uPortal, but wanted to get this out here to review the approach.

The gradle jar appeared to be missing, so I added it in.

### CODE REVIEW

**Comment by cbeach47** [comment]
@ChristianMurphy - is there a way to leave in console logging statements (behind the debug flag) without the linter yelling at me? Or maybe I just need a better approach for logging for future investigations....

**Comment by cbeach47** [comment]
Looks good in Edge, FF, and Chrome

**Comment by ChristianMurphy** [comment]
> is there a way to leave in console logging statements (behind the debug flag) without the linter yelling at me? Or maybe I just need a better approach for logging for future investigations

@cbeach47 do you have direct access to the running portal?
Or are you indirectly collecting issues?

If the former https://developers.google.com/web/tools/chrome-devtools/javascript/ and https://developer.mozilla.org/en-US/docs/Tools/Debugger both avoid needing to add extra logs, and also provide more detailed and helpful information for tracing.
If the latter, https://github.com/pimterry/loglevel offers a way to control which log levels are enabled and disabled.

File: @uportal/context-menu-navigation/src/components/ContextMenu.vue [hunk 1]
```diff
@@ -169,14 +169,59 @@ export default {
...
+            } else {
+                return this.selected;
+            }
+        },
+        cleanWidgetTemplate() {
+            const escapedXmlMapping = {
+                '&lt;': '<',
+                '&gt;': '>',
+                '&quot;': '"',
+                '&amp;': '&'
```
Comment (ChristianMurphy) [inline @uportal/context-menu-navigation/src/components/ContextMenu.vue:212]:
```suggestion
                '&lt;': '<',
                '&gt;': '>',
                '&quot;': '"',
                '&apos; "'",
                '&amp;': '&'
```

File: @uportal/context-menu-navigation/src/components/ContextMenu.vue [hunk 2]
```diff
@@ -169,14 +169,59 @@ export default {
...
+            };
+            // Assumes portletContent is truthy, thus cid will be defined
+            const cid = this.contentItemDetails;
+            if (cid && cid.parameters && cid.parameters.widgetTemplate) {
```
Comment (ChristianMurphy) [inline @uportal/context-menu-navigation/src/components/ContextMenu.vue:216]: You may want to try this out, not sure if babel is updated enough to support it yet, but it is easier to read if it does
```suggestion
            if (this.contentItemDetails?.parameters?.widgetTemplate) {
```

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining

File: @uportal/context-menu-navigation/src/components/ContextMenu.vue [hunk 3]
```diff
@@ -169,14 +169,59 @@ export default {
...
+            } else {
+                return this.selected;
+            }
+        },
+        cleanWidgetTemplate() {
+            const escapedXmlMapping = {
+                '&lt;': '<',
+                '&gt;': '>',
+                '&quot;': '"',
+                '&apos; "'",
```
Comment (ChristianMurphy) [inline @uportal/context-menu-navigation/src/components/ContextMenu.vue:212]: whoops missed a couple chars
```suggestion
                '&apos;': "'",
```

Comment (cbeach47) [reply @uportal/context-menu-navigation/src/components/ContextMenu.vue:212]: lol, :)  got that when pulling the previous!

**Comment by cbeach47** [comment]
@ChristianMurphy - I can see both ways to access uPortal.  Based on our discussion earlier, I implemented [debug](https://www.npmjs.com/package/debug#browser-support) as a prototype on how we can unify uPWC ( in a general term) logging.  It fit precisely with what I was after. With a live Prod install, an end user should just need to use their browser console and enter `localStorage.debug="up:*"` and refresh the page to launch the verbose logging.

So:
- up:context-nav-menu
- up:waffle-menu (future)
- up:dashboard-carousel (future)
- etc....

Just in my testing, I realized other components, like `sockjs` is already using `debug`, so it's a more widely used way to enable logging.

**Comment by cbeach47** [comment]
The commit linter doesn't like when I 'committed' your changes via the github UX.  Do you know of a clean way to have the linter ignore these two commits, and I'll squash the code as a final step?

image

**Comment by ChristianMurphy** [comment]
> UX. Do you know of a clean way to have the linter ignore these two commits, and I'll squash the code as a final step?


Not off the top of my head.
It may make sense to move the lint step in Travis CI to be last, to guarantee everything else runs.

**Review by bjagg — APPROVED** [review]

**Pull request merged**